### DISCUSSION TITLE
Is it possible to read/write atom outside React component?

### DISCUSSION THREAD

**Discussion by dvakatsiienko** [pmndrs/jotai#1478] [Q&A]
I use a library that exposes the data using non-react-component callbacks.

I need to save a ref to HTML element inside of that callback.
But the problem is that it is not a React Component.

It is possible to write to Jotai atom from a regular function?

Example:

```typescript
const elementlRefAtom = atom<HTMLElement | null>(null);

beforeEventEdit(event) { // ← NOT a React Component
  // event.elementRef — holds a ref to a target HTML element

  const [, setElementlRefAtom] = useAtom(elementlRefAtom); // ← breaks, because enclosing function is not a React Component
  setElementlRefAtom(event.elementRef);
},
```

**Comment by dai-shi** [comment]
Jotai atoms are contextual, which means not globally accessible. It's the same as `useState`.
To support global state usage, `unstable_createStore` #922 is added, but it's still experimental.
